buildSrc/src/main/kotlin/KotlinVersion.kt

```diff
@@ -0,0 +1,11 @@
+@file:JvmName("KotlinVersion")
+
+import org.gradle.api.*
+
+fun isKotlinVersionAtLeast(kotlinVersion: String, atLeastMajor: Int, atLeastMinor: Int): Boolean {
+    val (major, minor) = kotlinVersion
+        .split('.')
+        .take(2)
+        .map { it.toInt() }
+    return major == atLeastMajor && minor >= atLeastMinor
```

Comment:
If conditional removal of JS/Legacy persists up until migration to 2.0.0 (or 2.0.0-dev), JS/Legacy will be enabled back after the switch to 2.0.0 as `isKotlinVersionAtLeast` considers `2.0 < 1.9` right now due to equality comparison on major version. I suggest the following tweak for this line:
```kotlin
return when {
    major > atLeastMajor -> true
    major < atLeastMajor -> false
    else -> minor >= atLeastMinor
}
```